### PR TITLE
Added routine to access components without specifying if wrapped

### PR DESCRIPTION
**Pull request by michalpelka**
A lot of components in this Gem are created wrapped in GenericComponentWrapper. Those need to be accessed differently in game mode and editor. `GetGameOrEditorComponent` was added to help with that.

Signed-off-by: Michał Pełka <michalpelka@gmail.com>

### CODE REVIEW

File: Code/Source/Camera/CameraSensor.cpp [hunk 1]
```diff
@@ -132,7 +132,7 @@ namespace ROS2
         AZ::Transform inverse = cameraPose.GetInverse();
         m_view->SetWorldToViewMatrix(AZ::Matrix4x4::CreateFromQuaternionAndTranslation(inverse.GetRotation(), inverse.GetTranslation()));
 
-        size_t userId = AZ::Render::FrameCaptureRequests::s_InvalidFrameCaptureId;
```
Comment (adamdbrw) [inline Code/Source/Camera/CameraSensor.cpp]: Probably should not be here? It seems that this is an unrelated change.

File: Code/Source/ROS2GemUtilities.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        /// Retrieve component from entity given by a pointer. It is a way to get game components and wrapped components.
+        /// We should use that that we are not sure if we access e.g. ROS2FrameComponent in game mode or from Editor
+        /// \param entity pointer to entity e.g. with GetEntity()
+        /// \return pointer to component with type T
+
+        template<class ComponentType>
+        ComponentType* GetGameOrEditorComponent(const AZ::Entity* entity)
+        {
+            AZ_Assert(entity, "Called with empty entity");
+            ComponentType* component = entity->FindComponent<ComponentType>();
```
Comment (adamdbrw) [inline Code/Source/ROS2GemUtilities.h:33]: Is there reasoning why we are looking in this order?

Comment (adamdbrw) [reply Code/Source/ROS2GemUtilities.h:33]: I was wondering whether we could check whether we are in Game mode or Editor mode and retrieve based on that. Would that work?

**Review by adamdbrw — COMMENTED** [review]
Important fix. Could be accepted as it is, but please take a look at the comments first.

File: Code/Source/ROS2GemUtilities.h (hunk 2 above)
Comment (michalpelka) [reply Code/Source/ROS2GemUtilities.h:33]: Yes, the logic is that game mode is more time critical.

File: Code/Source/Camera/CameraSensor.cpp (hunk 1 above)
Comment (michalpelka) [reply Code/Source/Camera/CameraSensor.cpp]: Done

**Review by adamdbrw — APPROVED** [review]
